arangod/V8Server/v8-collection.cpp

```diff
@@ -72,6 +72,40 @@
 #include <velocypack/Slice.h>
 #include <velocypack/velocypack-aliases.h>
 
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief retrieves a collection from a V8 argument
+////////////////////////////////////////////////////////////////////////////////
+std::shared_ptr<arangodb::LogicalCollection> GetCollectionFromArgument(
+    TRI_vocbase_t& vocbase,
+    v8::Handle<v8::Value> const val
+) {
+  if (arangodb::ServerState::instance()->isCoordinator()) {
+    try {
+      auto* ci = arangodb::ClusterInfo::instance();
+
+      return ci
+        ? ci->getCollection(vocbase.name(), TRI_ObjectToString(val)) : nullptr;
+    } catch (...) {
+      // NOOP
+    }
+
+    return nullptr; // not found
+  }
+
+  // number
+  if (val->IsNumber() || val->IsNumberObject()) {
```

Comment:
don't you have to support lookups by plan ID in the coordinator as well ?

Reply:
this was moved from line 1977 and combined with lines 2268/2351